### PR TITLE
feat: Exclude configs from actions while committing to git repo

### LINKED ISSUE #11997: [Feature]: Exclude config values from actions while committing to git repo
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- Exclude the app slug from git repo as the imported app from git repo will have different name 
- For actions created from datasources of type SAAS, API we need to exclude the config values from committing to git repo
- For git import provide an option for the user to populate these values during the import. Needs discussion. 

Reference - https://theappsmith.slack.com/archives/CPQNLFHTN/p1647497578863639

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
Exclude the app slug from git repo as the imported app from git repo will have different name.
For actions created from datasources of type SAAS, API we need to exclude the config values from committing to git repo

Fixes #11997 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Locally

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by abhvsn** [comment]
I'll let @nidhi-nair to review this PR as I don't have the complete context here 🙂

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog2.java [hunk 1]
```diff
@@ -644,7 +644,25 @@ public void updateIsConfiguredFlagForAllTheExistingDatasources(MongockTemplate m
         }
     }
 
-    @ChangeSet(order = "005", id = "set-application-version", author = "")
+    @ChangeSet(order = "005", id = "add-isConfigStoredAtDataSource-for-plugins", author = "")
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog2.java:647]: Let's maintain ordering here?

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java [hunk 2]
```diff
@@ -1415,7 +1416,7 @@ public void pullChanges_upstreamChangesAvailable_pullSuccess() throws IOExceptio
         gitStatusDTO.setBehindCount(0);
         gitStatusDTO.setIsClean(true);
 
-        Mockito.when(gitFileUtils.saveApplicationToLocalRepo(Mockito.any(Path.class), Mockito.any(ApplicationJson.class), Mockito.anyString()))
+        Mockito.when(gitFileUtils.saveApplicationToLocalRepo(Mockito.any(Path.class), Mockito.any(ApplicationJson.class), Mockito.anyString(), Mockito.anyList()))
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java:1419]: Could we please add a test for `true`, `false` and `null` values here?

**Comment by nidhi-nair** [comment]
@AnaghHegde, for the remote plugin migration, we'll also have Twilio and a bunch of Hubspot plugins on prod. Could you please limit that to Airtable. Also let's talk to @hiteshjoshi and get a clear set of guidelines about when we're supposed to mark this value as true or false from a product perspective? We can use Twilio and Hubspot as examples to figure this out. This will help our SaaS (remote) team set this up independently.

Also, looks like our default behaviour for this is that we do copy over the config. Is that intended? If one of the plugins skips on this config because of an oversight, what behaviour do we want to default to?

**Comment by AnaghHegde** [comment]
@nidhi-nair For the default behaviour yes we do copy over the config as it does not have any sensitive data and does not affect the import flow.For Twilio and Hubspot we can have a call to discuss this in detail.

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java (hunk 2 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/GitServiceTest.java:1419]: It won't make sense to include this here, as we are mocking the implementation here. But i am writing test for this in `GitFileUtilsTest` class, i will push this.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Plugin.java [hunk 3]
```diff
@@ -82,4 +82,6 @@
     @Transient
     Map<String, String> templates;
 
+    Boolean isConfigStoredAtDataSource;
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Plugin.java:85]: Nit: Can we use lowercase for datasource? This is to maintain consistency with how we refer to datasources across Appsmith.

**Review by nidhi-nair — CHANGES_REQUESTED** [review]
Just a tiny change in the name that'll help with maintenance

File: app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Plugin.java (hunk 3 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Plugin.java:85]: @nidhi-nair If you look at the [L.72 ](https://github.com/appsmithorg/appsmith/blob/release/app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Plugin.java#L72) we have used it in capital words.

Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Plugin.java]: Thanks for catching this, we should fix this reference as well!

**Comment by mohanarpit** [comment]
@AnaghHegde Since the linked issue has been closed because we don't wish to follow the proposed implemenetation plan, is this PR still relevant? Can we close this PR?

**Comment by AnaghHegde** [comment]
Closing this PR as we have decided to not to implement this feature.
